pkg/kubelet/nodestatus/setters.go

```diff
@@ -121,6 +123,10 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 		}
 
 		if externalCloudProvider {
+			// If --cloud-provider=external delegate the node address configuration to the cloud provider.
+			if !utilfeature.DefaultFeatureGate.Enabled(features.InitializeNodeAddressesCloudProviderExternal) {
+				return nil
```

Comment:
Want to make sure we ruled out all the initial reasons for why we added this check. From digging through https://github.com/kubernetes/kubernetes/pull/75229 it seems like the primary use-case was addressing kubeadm self-hosting where you need an initial node address for the bootstrap to work. From other comments it sounds like this is no longer supported? 

I also found https://github.com/kubernetes/cloud-provider-vsphere/issues/299, which implies private cloud providers like vSphere might be relying on this behavior.

Reply:
Hi Andrew, do you remember why cloud-provider-vsphere needs it at the first place?
The error in CPI looks like is from kubernetes https://github.com/saschagrunert/kubernetes/blob/e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529/pkg/master/master.go#L461, the NoMatchError. I assume it's because this function got called in CPI somehow?

Reply:
Maybe that's why we added this logic, it's hard to debug issues in CPI if we don't have this logic. Since we can't fetch pod's log directly

Reply:
as in https://github.com/kubernetes/kubernetes/issues/120720#issuecomment-1727025248, I think that the problem is that node should not be ready if there are no addresses, and not compensate that by adding any address that is present in the node temporary

Reply:
maybe I should add the other PR as a separate commit here https://github.com/kubernetes/kubernetes/pull/120753 🤔